src/textual/widget.py

```diff
@@ -514,38 +514,50 @@ def scroll_to_widget(self, widget: Widget, *, animate: bool = True) -> bool:
             bool: True if the scroll position changed, otherwise False.
         """
 
-        try:
-            widget_region = widget.content_region
-            container_region = self.content_region
-        except errors.NoWidget:
-            return False
-
-        if widget_region in container_region:
-            # Widget is visible, nothing to do
-            return False
+        node = widget.parent
+        child = widget
+        while node:
+            try:
+                widget_region = child.content_region
+                container_region = node.content_region
+            except (errors.NoWidget, AttributeError):
+                return False
+
+            if widget_region in container_region:
+                # Widget is visible, nothing to do
+                child = node
+                node = node.parent
+                continue
+
+            # We can either scroll so the widget is at the top of the container, or so that
+            # it is at the bottom. We want to pick which has the shortest distance
+            top_delta = widget_region.origin - container_region.origin
+
+            bottom_delta = widget_region.origin - (
+                container_region.origin
+                + Offset(0, container_region.height - widget_region.height)
+            )
 
-        # We can either scroll so the widget is at the top of the container, or so that
-        # it is at the bottom. We want to pick which has the shortest distance
-        top_delta = widget_region.origin - container_region.origin
+            if widget_region.width > container_region.width:
+                delta_x = top_delta.x
+            else:
+                delta_x = min(top_delta.x, bottom_delta.x, key=abs)
 
-        bottom_delta = widget_region.origin - (
-            container_region.origin
-            + Offset(0, container_region.height - widget_region.height)
-        )
+            if widget_region.height > container_region.height:
+                delta_y = top_delta.y
+            else:
+                delta_y = min(top_delta.y, bottom_delta.y, key=abs)
 
-        if widget_region.width > container_region.width:
-            delta_x = top_delta.x
-        else:
-            delta_x = min(top_delta.x, bottom_delta.x, key=abs)
+            node.scroll_relative(
+                delta_x or None, delta_y or None, animate=animate, duration=0.2
+            )
 
-        if widget_region.height > container_region.height:
-            delta_y = top_delta.y
-        else:
-            delta_y = min(top_delta.y, bottom_delta.y, key=abs)
+            if node == self:
+                break
+            child = node
+            node = node.parent
 
-        return self.scroll_relative(
-            delta_x or None, delta_y or None, animate=animate, duration=0.2
-        )
+        # TODO: Return boolean value
```

Comment:
shouldn't this TODO be addressed before we merge the PR?
(unless it's a complicated value to compute with this new version of the method, of course :-)